Sources/Shared/Protocols/MappableOnce.swift

```diff
@@ -0,0 +1,5 @@
+import Sugar
+
+public protocol MappableOnce {
+  init?(_ map: JSONDictionary) throws
```

Comment:
Just think that one protocol is better that 2 for public API, also the name `MappableOnce` is a bit weird, in both cases it's initialiser, so you can map only once :smile: